### PR TITLE
TE-16.2 Encap Frr Test - fix for using l2 rules and sequenceID

### PR DESCRIPTION
**Pull request by suredhar**
1. Added back code to use deviation for default class by using L2 rules.
2. Added a function to generate sequence IDs off a base value to avoid out-of-order rule sequences in a policy-map configuration.
3. Added a deviation for the gRIBI IPv4 route. The device does not support a MAC address, only a next-hop entry.
4. Added the following deviations:
    -> gribi_mac_override_with_static_arp
    -> interface_ref_interface_id_format
   -> pf_require_match_default_rule

### CODE REVIEW

**Review by AmrNJ — CHANGES_REQUESTED** [review]
Fails at speed setting.

File: feature/gribi/otg_tests/encap_frr/encap_frr_test.go [hunk 1]
```diff
@@ -496,6 +621,53 @@ func staticARPWithMagicUniversalIP(t *testing.T, dut *ondatra.DUTDevice) {
 	sb.Set(t, dut)
 }
 
+func programAftWithDummyIp(t *testing.T, dut *ondatra.DUTDevice, args *testArgs) {
```
Comment (NishaSadhasivam-work) [inline feature/gribi/otg_tests/encap_frr/encap_frr_test.go:624]: rename programAftWithDummyIp to programAftWithDummyIP

File: feature/gribi/otg_tests/encap_frr/encap_frr_test.go [hunk 2]
```diff
@@ -529,6 +701,8 @@ func configureGribiRoute(ctx context.Context, t *testing.T, dut *ondatra.DUTDevi
 			fluent.NextHopGroupEntry().WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
 				WithID(15).AddNextHop(16, 1),
 		)
+	} else if deviations.GRIBIMACOverrideWithStaticARP(dut) {
+		programAftWithDummyIp(t, dut, args)
```
Comment (NishaSadhasivam-work) [inline feature/gribi/otg_tests/encap_frr/encap_frr_test.go:705]: rename programAftWithDummyIp to programAftWithDummyIP

**Review by NishaSadhasivam-work — APPROVED** [review]
LGTM!

**Review by NishaSadhasivam-work — APPROVED** [review]
LGTM!

**Review by AmrNJ — APPROVED** [review]

**Pull request merged**